### PR TITLE
W1 - D5 - Activity - Fragment

### PR DESCRIPTION
**Pull request by at-dinhtruong**
W1 d5 activity fragment #10

### CODE REVIEW

File: app/src/at-dinhtruong/java/asiantech/internship/summer/LoginFragment.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
+                fragmentTransaction.setCustomAnimations(R.anim.right_to_left1, R.anim.right_to_left2, R.anim.left_to_right1, R.anim.left_to_right2);
+                fragmentTransaction.add(R.id.fragment_container, new SignUpFragment());
+                fragmentTransaction.addToBackStack(new Fragment().getClass().getSimpleName());
+                fragmentTransaction.commit();
+                break;
+            }
+            case R.id.btnLogin: {
+                String mEmail = mEditTextEmail.getText().toString();
+                String mPassWord = mEditTextPassWord.getText().toString();
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/LoginFragment.java]: @at-dinhtruong : mPassWord -> passWord

File: app/src/at-dinhtruong/java/asiantech/internship/summer/ShowInformationActivity.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package asiantech.internship.summer;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+public class ShowInformationActivity extends AppCompatActivity {
+    private TextView mTextViewEmail;
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/ShowInformationActivity.java]: @at-dinhtruong : mTextViewEmail -> mTvEmail

File: app/src/at-dinhtruong/java/asiantech/internship/summer/ShowInformationActivity.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private TextView mTextViewPassword;
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/ShowInformationActivity.java]: @at-dinhtruong : mTextViewPassword -> mTvPassword

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+
+import asiantech.internship.summer.utils.Validate;
+
+public class SignUpFragment extends Fragment implements View.OnClickListener {
+    private Button mButtonSignUp;
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mButtonSignUp -> mBtnSignUp

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    private EditText mEditTextEmail;
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mEditTextEmail -> mEdtEmail

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    private EditText mEditTextPassWord;
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mEditTextPassWord -> mEdtPassWord

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    private EditText mEditTextConFirmPassWord;
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mEditTextConFirmPassWord -> mEdtConFirmPassWord

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        mEditTextEmail = view.findViewById(R.id.edtEmail);
+        mEditTextPassWord = view.findViewById(R.id.edtPassword);
+        mEditTextConFirmPassWord = view.findViewById(R.id.edtConfirmPassword);
+        mButtonSignUp.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        Validate validate = new Validate();
+        String mEmail = mEditTextEmail.getText().toString();
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mEmail -> email

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        String mPassword = mEditTextPassWord.getText().toString();
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mPassword -> password

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        String mConfirmPassword = mEditTextConFirmPassWord.getText().toString();
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : mConfirmPassword -> confirmPassword

File: app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+                if (validate.isValidEmail(mEmail) && validate.isValidPassword(mPassword)) {
+                    if (mPassword.equals(mConfirmPassword)) {
+                        if (getActivity() instanceof LoginActivity) {
+                            ((LoginActivity) getActivity()).setTitleToolBar(getString(R.string.login));
+                            ((LoginActivity) getActivity()).setButtonBack(0);
+                        }
+                        FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
+                        fragmentTransaction.setCustomAnimations(R.anim.right_to_left1, R.anim.right_to_left2);
+                        fragmentTransaction.add(R.id.fragment_container, new LoginFragment());
+                        fragmentTransaction.addToBackStack(new Fragment().getClass().getSimpleName());
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/SignUpFragment.java]: @at-dinhtruong : new Fragment().getClass().getSimpleName() ->  LoginFragment.class.getSimpleName()

File: app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validate {
+    public boolean isValidPassword(String mPassword) {
+        Pattern patternPassword;
+        Matcher matchePassword;
+        final String PASSWORD_PATTERN = "^[a-zA-Z0-9]{7,}$";
+        patternPassword = Pattern.compile(PASSWORD_PATTERN);
+        matchePassword = patternPassword.matcher(mPassword);
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java]: @at-dinhtruong : matchePassword -> matcherPassword

File: app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
+package asiantech.internship.summer.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validate {
+    public boolean isValidPassword(String mPassword) {
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java]: @at-dinhtruong : mPassword -> password

File: app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+        Pattern patternPassword;
+        Matcher matchePassword;
+        final String PASSWORD_PATTERN = "^[a-zA-Z0-9]{7,}$";
+        patternPassword = Pattern.compile(PASSWORD_PATTERN);
+        matchePassword = patternPassword.matcher(mPassword);
+        return matchePassword.matches();
+    }
+
+    public boolean isValidEmail(String mEmail) {
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java]: @at-dinhtruong : mEmail -> email

File: app/src/at-dinhtruong/res/layout/activity_login.xml [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+        android:layout_height="50dp"
+        android:background="@color/colorOrange">
+
+        <ImageView
+            android:id="@+id/imgBack"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content" />
+
+        <TextView
+            android:id="@+id/toolBarTitle"
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/res/layout/activity_login.xml]: @at-dinhtruong : toolBarTitle -> tvToolBarTitle

File: app/src/at-dinhtruong/res/layout/fragment_signup.xml [hunk 16]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/fragment_login"
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/res/layout/fragment_signup.xml]: @at-dinhtruong : fragment_login -> llContainer

File: app/src/at-dinhtruong/res/values/strings.xml [hunk 17]
```diff
@@ -1,3 +1,12 @@
 <resources>
     <string name="app_name">Intern18Summer</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="passwordFormatWrong">Password format Wrong!</string>
+    <string name="confirmPasswordWrong">Confirm Password Wrong!</string>
+    <string name="signup">Signup?</string>
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/res/values/strings.xml]: @at-dinhtruong : Signup? -> Signup

Comment (at-dinhtruong) [reply app/src/at-dinhtruong/res/values/strings.xml]: ở đề chữ "Signup?" anh ạ.

**Comment by at-dinhtruong** [comment]
I fixed your comment!

File: app/src/at-dinhtruong/res/values/strings.xml (hunk 17 above)
Comment (at-hanhnguyen2) [reply app/src/at-dinhtruong/res/values/strings.xml]: @at-dinhtruong : but you set this text for title toolbar, text title toolbar should only "Signup"

File: app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java [hunk 18]
```diff
@@ -0,0 +1,25 @@
+package asiantech.internship.summer.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validate {
+    private final String PASSWORD_PATTERN = "^[a-zA-Z0-9]{7,}$";
```
Comment (at-hanhnguyen2) [inline app/src/at-dinhtruong/java/asiantech/internship/summer/utils/Validate.java]: @at-dinhtruong : change to private static final String PASSWORD_PATTERN to fix warning

File: app/build.gradle [hunk 19]
```diff
@@ -56,12 +56,12 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'com.android.support:appcompat-v7:27.1.1'
-    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Comment (at-dev-support-01) [inline app/build.gradle]: WARNING: [GradleDependency] A newer version of com.android.support:support-v4 than 27.0.0 is available: 27.1.1
 This detector looks for usages of libraries where the version you are using is not the current stable release. Using older versions is fine, and there are cases where you deliberately want to stick with an older version. However, you may simply not be aware that a more recent version is available, and that is what this lint check helps find.

File: app/src/at-dinhtruong/res/menu/toolbar.xml [hunk 20]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/icBack"
+        android:icon="@drawable/ic_arrow_back_black_36dp"
+        android:title="Compose"
```
Comment (at-dev-support-01) [inline app/src/at-dinhtruong/res/menu/toolbar.xml]: WARNING: [HardcodedText] Hardcoded string "Compose", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait) you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

File: app/build.gradle [hunk 21]
```diff
@@ -64,4 +64,5 @@ dependencies {
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
     implementation 'de.hdodenhof:circleimageview:2.2.0'
     apply from: "../ci-java.gradle"
+    implementation 'com.android.support:support-v4:27.0.0'
```
Comment (at-dev-support-01) [inline app/build.gradle]: WARNING: [GradleDependency] A newer version of com.android.support:support-v4 than 27.0.0 is available: 27.1.1
 This detector looks for usages of libraries where the version you are using is not the current stable release. Using older versions is fine, and there are cases where you deliberately want to stick with an older version. However, you may simply not be aware that a more recent version is available, and that is what this lint check helps find.

**Pull request merged**